src/Illuminate/Collections/LazyCollection.php

```diff
@@ -556,6 +556,16 @@ public function isEmpty()
         return ! $this->getIterator()->valid();
     }
 
+    /**
+     * Determine if the collection contains a single element.
+     *
+     * @return bool
+     */
+    public function isSingle()
+    {
+        return $this->take(2)->count() === 1;
```

Comment:
why is `take(2)` necessary? :)

Reply:
@gocanto without it, the whole collection would have to be enumerated, which is a waste.

As soon as we know that there are 2, we know that it's not single!

Reply:
cool stuff.